src/main/scala/nl/knaw/dans/easy/multideposit/Main.scala

```diff
@@ -40,37 +40,34 @@ object Main extends DebugEnhancedLogging {
 
   def run(implicit settings: Settings): Try[Unit] = {
     MultiDepositParser.parse(multiDepositInstructionsFile)
-      .flatMap(getActions(_).reduce(_ compose _).run)
+      .flatMap(getActions(_).reduce(_ andThen _).run())
   }
 
-  def getActions(datasets: Datasets)(implicit settings: Settings): ListBuffer[Action] = {
+  def getActions(datasets: Datasets)(implicit settings: Settings): ListBuffer[Action[Unit]] = {
```

Comment:
An example of why we need this parameterized `Action` can already be found in the current code base. The `datamanager` that is provided in the command line arguments needs to be validated and it's email address needs to be queried from LDAP. Given that this is the same for all datasets in this multi-deposit, we only want to execute this query once. Currently we use a very Java-like way with a private (static) variable which is initialized to `null` and allocated with the real emailaddress as the query gets executed. This gives interesting challenges while testing (including private method invocation via reflection) and isn't really Scala-like.

The new implementation makes retrieving the datamanager's email into an atomic `Action`, which is only executed once in this `Main.scala`. It's result (the email address) is injected into the `AddPropertiesToDeposit` action by making the latter of type `Action[String => Unit]` and using the new `applyRight` operator.